src/Uri.php

```diff
@@ -652,7 +652,7 @@ private function filterFragment(string $fragment) : string
     /**
      * Filter a query string key or value, or a fragment.
      */
-    private function filterQueryOrFragment(string $value) : string
+    private function filterQueryOrFragment(string $value) : ?string
```

Comment:
This kind of change requires a backing test case.

Reply:
ok, I've fix a test case.

Reply:
We'd need an *additional* test case, not a change in existing ones